src/components/Initiatives/InitiativeDetailsPage/InitiativeTableCard/FailingComponentsTable.tsx

```diff
@@ -13,21 +13,45 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
+
+import { EmptyState, InfoCard, Table as BSTable, TableColumn } from '@backstage/core-components';
+
 import { InitiativeActionItem } from '../../../../api/types';
-import { Table, TableBody, TableCell, TableRow } from '@material-ui/core';
 import { groupByString } from '../../../../utils/collections';
 import { useDetailCardStyles } from '../../../../styles/styles';
 import { FailingComponentsTableRow } from './FailingComponentsTableRow';
-import { InfoCard } from '@backstage/core-components';
+import { defaultComponentRefContext } from '../../../../utils/ComponentUtils';
+import { humanizeAnyEntityRef } from '../../../../utils/types';
+import { Box } from '@material-ui/core';
+
+const columns: TableColumn[] = [{
+  field: 'all',
+  render: (data: { componentRef?: string, numRules?: number, actionItems?: InitiativeActionItem[] }) => {
+    return (
+      <FailingComponentsTableRow
+        componentRef={data.componentRef ?? ''}
+        actionItems={data.actionItems ?? []}
+        numRules={data.numRules ?? 0}
+      />
+    );
+  },
+  customFilterAndSearch: (filter, rowData: { serviceName?: string }) => {
+    return rowData.serviceName?.indexOf(filter) !== -1;
+  },
+  sorting: false,
+  title: '',
+}];
 
 interface FailingComponentsTableProps {
   actionItems: InitiativeActionItem[];
+  defaultPageSize?: number;
   numRules: number;
 }
 
 export const FailingComponentsTable = ({
   actionItems,
+  defaultPageSize = 15,
```

Comment:
This page size is somewhat arbitrary and I'm happy to hear feedback or suggestions!

Reply:
thoughts on adding a const for PAGE_SIZE and using it throughout so that we always use the same one?

Reply:
I think we actually want a different page size here than the other location (scorecard details page) because there are 2 tables here stacked on top of each other here and only 1 there.